source/fab/tools/flags.py

```diff
@@ -0,0 +1,78 @@
+##############################################################################
+# (c) Crown copyright Met Office. All rights reserved.
+# For further details please refer to the file COPYRIGHT
+# which you should have received as part of this distribution
+##############################################################################
+
+'''This file contains a simple Flag class to manage tool flags.
+It will need to be combined with build_config.FlagsConfig in a follow up
+PR.
+'''
+
+import logging
+from typing import List, Optional
+import warnings
+
+from fab.util import string_checksum
+
+
+class Flags(list):
+    '''This class represents a list of parameters for a tool. It is a
+    list with some additional functionality.
+
+    TODO #22: This class and build_config.FlagsConfig should be combined.
+
+    :param list_of_flags: List of parameters to initialise this object with.
+    '''
+
+    def __init__(self, list_of_flags: Optional[List[str]] = None):
+        self._logger = logging.getLogger(__name__)
+        super().__init__()
+        if list_of_flags:
+            self.extend(list_of_flags)
+
+    def checksum(self) -> str:
+        """
+        :returns: a checksum of the flags.
+
+        """
+        return string_checksum(str(self))
+
+    def remove_flag(self, remove_flag: str, has_parameter: bool = False):
+        '''Removes all occurrences of `remove_flag` in flags`.
+        If has_parameter is defined, the next entry in flags will also be
+        removed, and if this object contains this flag+parameter without space
+        (e.g. `-J/tmp`), it will be correctly removed. Note that only the
+        flag itself must be specified, you cannot remove a flag only if a
+        specific parameter is given (i.e. `remove_flag="-J/tmp"` will not
+        work if this object contains `[...,"-J", "/tmp"]`).
+
+        :param remove_flag: the flag to remove
+        :param has_parameter: if the flag to remove takes a parameter
+        '''
+        i = 0
+        flag_len = len(remove_flag)
+        while i < len(self):
```

Comment:
I already have a ticket for flag improvements - https://github.com/hiker/fab/issues/22 . Esp. fixes, since this PR badly introduces a new flag object ... my apologies, when I saw how big this PR became, I left the flag pretty basic, to be fixed later). For improved visibility, I've opened #313 (which also references our internal ticket)